x-pack/plugins/security_solution/public/common/components/charts/donutchart.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiLink, EuiText, EuiTextColor } from '@elastic/eui';
+import React, { useContext } from 'react';
+
+import {
+  Chart,
+  Datum,
+  Partition,
+  Settings,
+  PartitionLayout,
+  defaultPartitionValueFormatter,
+} from '@elastic/charts';
+import styled from 'styled-components';
+import { ThemeContext } from './donut_theme_context';
+import { useTheme } from './common';
+import { DraggableLegend } from './draggable_legend';
+import { LegendItem } from './draggable_legend_item';
+import { DonutChartEmpty } from './donutchart_empty';
+
+export const NO_LEGEND_DATA: LegendItem[] = [];
+
+interface DonutChartData {
+  key: string;
+  value: number;
+  group?: string;
+  label?: string;
+}
+
+export interface DonutChartProps {
+  data: DonutChartData[] | null | undefined;
+  fillColor: (d: Datum) => string;
```

Comment:
The fillColor prop is not the same as the Chart type, it should be `fillColor: string | NodeColorAccessor;`.
You may want to export it `export type FillColor = string | NodeColorAccessor;` if you need it outside this component.